### ISSUE TITLE
DeDRM Progress mod for support 64bit

### ISSUE THREAD

**Issue by SLboat** [apprenticeharper/DeDRM_tools#925]
thank you the great project!my english bad, so i just go the pointer.

this app have some problem at 10.15, we cant dedrm lot file anymore, as i deep into, found [DeDRM Progress]is a very old project,so i upgrade it to support new system with 64bit.

here is the source
[DeDRM Progress Sources(mod for support catlina).zip](https://github.com/apprenticeharper/DeDRM_tools/files/3782690/DeDRM.Progress.Sources.mod.for.support.catlina.zip)

end here is the compiled version
[DeDRM Progress(64bit).zip](https://github.com/apprenticeharper/DeDRM_tools/files/3782694/DeDRM.Progress.64bit.zip)

**Comment by ZolaLa9** [comment]
Awesome! Thanks!

Surprisingly, the DeDRM app still works for me without any modifications or replacement of DeDRM Progress. 

What does DeDRM Progress do?

**Comment by ElleKayEm** [comment]
DeDRM Progress was the piece of the Mac app that was reported as still being 32-bit.

**Comment by SLboat** [comment]
Yes indeed, it use to process batch file when drag to dedrm 


From SLboat

**Comment by ZolaLa9** [comment]
> DeDRM Progress was the piece of the Mac app that was reported as still being 32-bit.

Yes, I'm very well aware of that. I've been active on that thread and had tried to convert DeDRM Progress to 64-bit when I thought it was just an AppleScript. 

What is curious to me is that DeDRM works just fine for me on Catalina, despite having this piece of 32-bit code. I'm wondering what DeDRM Progress does since I am able to use the original DeDRM app without modifying or replacing DeDRM Progress. And yet, it stops others from success unless the original DeDRM Progress is replaced with a 64-bit version. 

> Yes indeed, it use to process batch file when drag to dedrm 

I don't understand what you mean by "process batch file." When I drag an ebook file to DeDRM it works fine on Catalina, even with the original DeDRM Progress. What is different between what I'm doing and what you're doing? Thanks

**Comment by SLboat** [comment]
You need drag multiple file not only one file:)


From SLboat

**Comment by SLboat** [comment]
When drag one file (like azw3),it will just dedrm it.

But when drag 2 or 3 files together to de-dedrm  ,it will call dedrm progress show the progress of decrypt, in this situation, if not update dedrm progress, it will tell you no ebook found.


From SLboat

**Comment by ZolaLa9** [comment]
Excellent! Thank you, **SLboat,** for explaining and thank you for compiling 64-bit version :D

**Comment by SLboat** [comment]
Glad I’m can help:)


From SLboat

**Comment by caramdache** [comment]
> just replace the old one at Resources folder it will work

@SLboat, I just tried that, but DeDRM is still marked as not compatible and it will not run.

I am on 10.15.4

**Comment by djproben** [comment]
Is there any update on this? Any version of DeDRM that will work with Catalina?

**Comment by ElleKayEm** [comment]
I don't have a Mac myself, but I've seen others using the current version on Catalina.

**Comment by ZolaLa9** [comment]
> Is there any update on this? Any version of DeDRM that will work with Catalina?

This issue (DeDRM Progress) just affects the standalone app. 

If you use the latest Calibre plugin, you should be okay.
